pulsar-metadata/src/test/java/org/apache/pulsar/metadata/MetadataCacheTest.java

```diff
@@ -611,7 +611,7 @@ public static void assertEqualsAndRetry(Supplier<Object> actual,
     public static boolean retryStrategically(Predicate<Void> predicate, int retryCount, long intSleepTimeInMillis)
             throws Exception {
         for (int i = 0; i < retryCount; i++) {
-            if (predicate.test(null) || i == (retryCount - 1)) {
```

Comment:
Interesting, not sure why adding `i == (retryCount - 1)` before

Reply:
Here has another one. It's very strange behavior.
https://github.com/apache/pulsar/blob/b17f86c794713e78a071652c969acdf917fd0074/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockedPulsarServiceBaseTest.java#L443

Reply:
I just take a look at https://github.com/apache/pulsar/pull/14373, think about why line.
I see your last comment with removing this.

Reply:
@Technoboy- Yes, the retryStrategically method is copied from [MockedPulsarServiceBaseTest.java](https://github.com/apache/pulsar/blob/b17f86c794713e78a071652c969acdf917fd0074/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/MockedPulsarServiceBaseTest.java#L443) , we should remove the check.